### DISCUSSION TITLE
Why is the response type returned when creating a new default middleware not its own response type?

### DISCUSSION THREAD

**Discussion by ZhangChengLin** [laravel/framework#51334] [Q&A]
Code involved:

https://github.com/laravel/framework/blob/eb04fb4e1722553bfe35e4c419805267b9c53ff8/src/Illuminate/Routing/Console/stubs/middleware.stub#L14

Scenes:

```

Laravel 11 

php artisan make:middleware SomeMiddleware

```

First of all, this question is a question from a newbie.

Why continue to use: \Symfony\Component\HttpFoundation\Response now
Shouldn't it be recommended to use Illuminate\Http\Response

Why hasn't it changed yet? Is it to maintain compatibility?
Or what are the benefits of doing so?

[Similar questions](https://github.com/laravel/framework/discussions?discussions_q=%5CSymfony%5CComponent%5CHttpFoundation%5CResponse)

**Comment by ghost** [comment]
namespace Illuminate\Http;

use Symfony\Component\HttpFoundation\Response as SymfonyResponse;

class Response extends SymfonyResponse

**Reply by ghost** [reply]
from this https://github.com/laravel/framework/discussions/50369
Illuminate\Http\RedirectResponse extends Symfony\Component\HttpFoundation\Response
Illuminate\Routing\Redirector does not.

So you question should be why is the dock block not up to date not why it is not returning Illuminate\Http\Response

**Reply by ZhangChengLin** [reply]
I don't understand what you said either.


I mean, when will he become like this?


`* @param \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response) $next
`

**Reply by ghost** [reply]
I can only guess that using Symfony response covers more than \Illuminate\Http\Response does, and as long as Symfony components are used in laravel, it is safer to put the "highest" parent there.

**Reply by ZhangChengLin** [reply]
I thought Laravel's response class extended SymfonyResponse
Things will be added.



[class Response extends SymfonyResponse](https://github.com/laravel/framework/blob/eb04fb4e1722553bfe35e4c419805267b9c53ff8/src/Illuminate/Http/Response.php#L15)

**Reply by ghost** [reply]
This is what I showed you here https://github.com/laravel/framework/discussions/51334#discussioncomment-9352943 

But even if it is extending it, there is still the possibility for the parent to be returned.
